### PR TITLE
Feature/abi spec

### LINKED ISSUE #5458: Implement abi spec
Problem

Currently we use `call-params` function to prepare the parameters for a contract call but this function is very naive and doesn't work with dynamic parameters.

Implementation

Write an `abi-spec` namespace with an encode function that takes the function signature and the parameters values and return the encoded data which can be used to call the contract.

Acceptance Criteria

Implementation should work for all exemples of the abi spec
https://solidity.readthedocs.io/en/develop/abi-spec.html

Future Steps

This issue is only about solving the encoding part. Similarly it would be desirable to implement the decoding part to handle return values.

<blockquote><div><strong><a href="https://solidity.readthedocs.io/en/develop/abi-spec.html">Contract ABI Specification — Solidity 0.4.25 documentation</a></strong></div></blockquote>

### PR DESCRIPTION
**Pull request by yenda**
fixes #5458

Implementation of the abi spec in cljs using web3-utils package for conversions

## testing notes

if the app runs on all platform (i.e get passed splash screen and can create account) it's good. This doesn't affect any code only dependencies

status: ready

### CODE REVIEW

**Comment by janherich** [comment]
Awesome work, clojure for the win !

**Review by janherich — APPROVED** [review]

File: src/status_im/utils/ethereum/abi_spec.cljs [hunk 1]
```diff
@@ -0,0 +1,269 @@
...
+;; - T[k] for any dynamic T and any k >= 0
+;; - (T1,...,Tk) if Ti is dynamic for some 1 <= i <= k
+(defn parse-param
+  "Takes a parsed parameter and returns a parameter that can
+  be encoded by associng the :dynamic? key for dynamic parameters
+  and recursively defining arrays"
+  [{::keys [type size array power] :as param}]
+  (if array
+    (let [{::keys [size]} (last array)]
+      {:type :array
```
Comment (goranjovic) [inline src/status_im/utils/ethereum/abi_spec.cljs]: map alignment

File: src/status_im/utils/ethereum/abi_spec.cljs [hunk 2]
```diff
@@ -0,0 +1,269 @@
...
+       :dynamic? (nil? size)
+       :array-of (parse-param (update param ::array butlast))})
+    (let [type (keyword type)
+          param {:type type
+                 :dynamic? (or (= type :string)
+                               (and (= type :bytes)
+                                    (nil? size)))
+                 :size size}]
```
Comment (goranjovic) [inline src/status_im/utils/ethereum/abi_spec.cljs]: map alignment

**Review by goranjovic — APPROVED** [review]

**Comment by antdanchenko** [comment]
The error ```'com.facebook.react.common.JavascriptException: undefined is not a valid argument for 'in' (evaluating '"StopIteration"in goog.global'), stack: @12:58248 d@2:768 n@2:339 t@2:262 global code@893:8 '``` is shown right after opening the app

0001screenshot

**Comment by yenda** [comment]
Android build works https://ci.status.im/job/status-react/job/combined/job/mobile-android/1369/artifact/StatusIm-180906-115101-69bb79-pr.apk

Waiting for desktop to update to react-native 0.56

<blockquote></blockquote>

**Pull request merged**